server/channels/app/content_flagging_setup_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetContentFlaggingGroupID(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("should register content flagging group", func(t *testing.T) {
+		groupID, err := th.App.GetContentFlaggingGroupID()
+		require.NoError(t, err)
+		require.NotEmpty(t, groupID)
+
+		// resetting cached value of group ID to ensure second function call goes to DB layer
+		contentFlaggingGroupID = ""
+
+		// should not update the group ID when called again
+		groupID2, err := th.App.GetContentFlaggingGroupID()
+		require.NoError(t, err)
+		require.Equal(t, groupID, groupID2)
+	})
+
+	t.Run("returns cached value if it exists", func(t *testing.T) {
+		contentFlaggingGroupID = "cached-group-id"
+		groupID, err := th.App.GetContentFlaggingGroupID()
+		require.NoError(t, err)
+		require.Equal(t, "cached-group-id", groupID)
+	})
+}
+
+func TestSetupContentFlaggingProperties(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	cleanup := func() {
+		contentFlaggingSetupOnce = sync.Once{} // Reset the once variable to allow re-running the setup
+
+		// Remove the system variable if it exists
+		_, err := th.App.Srv().Store().System().PermanentDeleteByName(contentFlaggingSetupDoneKey)
+		require.NoError(t, err)
+
+		// Remove the property fields
+		if contentFlaggingGroupID != "" {
+			propertyFields, err := th.App.Srv().propertyService.SearchPropertyFields(contentFlaggingGroupID, "", model.PropertyFieldSearchOpts{PerPage: 100})
+			require.NoError(t, err)
+
+			for _, propertyField := range propertyFields {
+				fieldErr := th.App.Srv().propertyService.DeletePropertyField(contentFlaggingGroupID, propertyField.ID)
+				require.NoError(t, fieldErr)
+			}
+		}
+
+		// Reset the cached group ID
+		contentFlaggingGroupID = ""
```

Comment:
Is it also necessary to remove from the DB?

Reply:
Removed the local variable `contentFlaggingGroupID ` so this is no longer applicable